[FSDP][BE] Only show state_dict log when the debug level is detail

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118196
* #118195
* #118197

As title

Differential Revision: [D53038704](https://our.internmc.facebook.com/intern/diff/D53038704/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

fegin committed:
[FSDP][BE] Only show state_dict log when the debug level is detail

As title

Differential Revision: [D53038704](https://our.internmc.facebook.com/intern/diff/D53038704/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP][BE] Only show state_dict log when the debug level is detail"


As title

Differential Revision: [D53038704](https://our.internmc.facebook.com/intern/diff/D53038704/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

fegin committed:
Update on "[FSDP][BE] Only show state_dict log when the debug level is detail"


As title

Differential Revision: [D53038704](https://our.internmc.facebook.com/intern/diff/D53038704/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Skylion007 commented on torch/distributed/fsdp/_debug_utils.py:
```diff
@@ -53,7 +53,9 @@ def profile(cls, profile_type: str) -> Iterator[None]:
-        logger.warning("%s %s", msg, str(cls.results))
+        # This cannot be combined with DETAIL distributed log
+        # as the profiling will be very incorrect.
+        logger.debug("%s %s", msg, str(cls.results))
```
```suggestion
        logger.debug("%s %s", msg, cls.results)
```
should work as well and be more efficient

wz337 reviewed (approved):
LG

rohan-varma commented on torch/distributed/fsdp/_debug_utils.py:
```diff
@@ -53,7 +53,9 @@ def profile(cls, profile_type: str) -> Iterator[None]:
     @classmethod
     def dump_and_reset(cls, msg: str) -> None:
-        logger.warning("%s %s", msg, str(cls.results))
+        # This cannot be combined with DETAIL distributed log
```
Could we at least do TORCH_DISTRIBUTED_DEBUG=INFO to do this then?

fegin committed:
Update on "[FSDP][BE] Only show state_dict log when the debug level is detail"


As title

Differential Revision: [D53038704](https://our.internmc.facebook.com/intern/diff/D53038704/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

fegin committed:
Update on "[FSDP][BE] Only show state_dict log when the debug level is detail"


As title

Differential Revision: [D53038704](https://our.internmc.facebook.com/intern/diff/D53038704/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]
